src/transformers/training_args.py

```diff
@@ -1186,6 +1213,25 @@ def __post_init__(self):
         if self.logging_strategy == IntervalStrategy.STEPS and self.logging_steps == 0:
             raise ValueError(f"logging strategy {self.logging_strategy} requires non-zero --logging_steps")
 
+        if (
+            self.logging_strategy == IntervalStrategy.STEPS
+            and self.logging_steps > 1
+            and self.logging_steps != int(self.logging_steps)
+        ):
+            raise ValueError(f"--logging_steps must be an integer if bigger than 1: {self.logging_steps}")
```

Comment:
Maybe here also convert the `logging_steps` to ints when they are bigger than 1 (same below for the two others).